projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -47,6 +43,24 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
     }, 0);
   }
 
+  private mapControls(
+    controlTransformationCallback?: (ctrl: AbstractControl, key: string) => any,
+    checkControlCallback: (ctrl: AbstractControl) => boolean = () => true,
+  ): any {
+    const controls: any = {};
+
+    for (const key in this.formGroup.controls) {
+      if (this.formGroup.controls.hasOwnProperty(key)) {
+        const control = this.formGroup.get(key);
+        if (control && checkControlCallback(control)) {
+          controls[key] = controlTransformationCallback ? controlTransformationCallback(control, key) : control;
+        }
+      }
+    }
+
+    return controls;
```

Comment:
I'm probably prematurely optimising here, but I feel this would be a great candidate for a `Proxy` as it would negate the need to iterate over the controls at all, rather it would be simply lookup complexity

Reply:
Would be glad to and I now definitely understand your point. It'd indeed be quite nice.
But after giving it a try, I have no idea how to do that on a property of an Angular component :thinking: Also wondering about the trade off usability VS eventually the need for a polyfill?

Reply:
Discussed IRL. Not going to do that now as it'd be quite a refactor to take into account the filter option AND because the eventual need for a polyfill is not something we want for devs. We might try to improve perfs later but did the best already by using a for loop instead of a map.